### PR TITLE
letting center_position be None

### PR DESCRIPTION
**Pull request by mganahl**
Let `center_position` be `None`, indicating that the MPS is not canonical

### CODE REVIEW

File: tensornetwork/matrixproductstates/base_mps.py [hunk 1]
```diff
@@ -72,7 +72,8 @@ def __init__(self,
         contractions. Available backends are currently 'numpy', 'tensorflow',
         'pytorch', 'jax'
     """
-    if center_position < 0 or center_position >= len(tensors):
+    if (center_position is not None) and (center_position < 0 or
+                                          center_position >= len(tensors)):
       raise ValueError(
           'center_position = {} not between 0 <= center_position < {}'.format(
               center_position, len(tensors)))
```
Comment (Thenerdstation) [inline tensornetwork/matrixproductstates/base_mps.py:79]: Update to error message to allow `None`

**Comment by mganahl** [comment]
added more testing (mostly covering raise statements)

File: tensornetwork/matrixproductstates/base_mps_test.py [hunk 2]
```diff
@@ -71,6 +71,18 @@ def test_backend_initialization(backend):
...
+def test_backend_initialization_raises(backend):
+  be = backend_factory.get_backend(backend)
+  D, d, N = 10, 2, 10
+  tensors = [np.random.randn(1, d, D)] + [
+      np.random.randn(D, d, D) for _ in range(N - 2)
+  ] + [np.random.randn(D, d, 1)]
+  with pytest.raises(ValueError):
+    BaseMPS(tensors, center_position=N, backend=be)
+  with pytest.raises(ValueError):
+    BaseMPS(tensors, center_position=-1, backend=be)
```
Comment (Thenerdstation) [inline tensornetwork/matrixproductstates/base_mps_test.py:83]: Add `match=...`

File: tensornetwork/matrixproductstates/base_mps_test.py [hunk 3]
```diff
@@ -158,6 +170,9 @@ def test_position_raises_error(backend):
     mps.position(-1)
   with pytest.raises(ValueError):
     mps.position(11)
+  mps = BaseMPS(tensors, center_position=None, backend=backend)
+  with pytest.raises(ValueError):
+    mps.position(1)
```
Comment (Thenerdstation) [inline tensornetwork/matrixproductstates/base_mps_test.py:175]: `match=...`

**Pull request merged**